dashboard/modules/job/sdk.py

```diff
@@ -28,14 +28,33 @@
 
 
 class JobSubmissionClient(SubmissionClient):
+    """A local client for submitting and interacting with jobs on a remote cluster.
+
+    Submits requests over HTTP to the job server on the cluster using the REST API.
+    """
+
     def __init__(
         self,
         address: str,
-        create_cluster_if_needed=False,
+        create_cluster_if_needed: bool = False,
         cookies: Optional[Dict[str, Any]] = None,
         metadata: Optional[Dict[str, Any]] = None,
         headers: Optional[Dict[str, Any]] = None,
     ):
+        """Initialize a JobSubmissionClient and check the connection to the cluster.
+
+        Args:
+            address: The IP address and port of the head node.
+            create_cluster_if_needed: Indicates whether the cluster at the specified
+                address needs to already be running. Ray doesn't start a cluster
+                before interacting with jobs, but external job managers may do so.
+            cookies: Cookies to use when sending requests to the HTTP job server.
+            metadata: Arbitrary metadata to store along with all jobs.  New metadata
+                specified per job will be merged with the global metadata provided here
+                via a simple dict update.
+            headers: Headers to use when sending requests to the HTTP job server, used
+                for cases like authentication to a remote cluster.
```

Comment:
@edoakes are these intended to be public? I pulled the documentation from documentation of internal functions, but I'm not sure if it's accurate or makes sense to be user-facing.

Reply:
The `cookies` and `headers` one definitely should be public, the others might be a little confusing because they're only used by niche external integrations. I think it's ok to make them public.